### Задача 1
Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания `mu` с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

In [7]:
import numpy as np
from scipy import stats
import scipy.stats as sps
import math

s_0 = 16
p = 0.95
M = 80
n = 256

a = 1 - p
Q1 = M - (sps.norm.ppf(p + a/2))*s_0/math.sqrt(256)
Q2 = M + (sps.norm.ppf(p + a/2))*s_0/math.sqrt(256)

print(f'Доверительный интервал [{Q1};{Q2}]')

Доверительный интервал [78.04003601545995;81.95996398454005]


### Задача 2
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

In [8]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m = sample.mean()
n = 10
sigma = np.sqrt(((sample - m)**2).sum()/(n-1))
print ('𝜎 =', sigma)

𝜎 = 4.453463071962462


Табличное значение коэффициента Стьюдента

In [9]:
t_int = 2.8214

Проверим, что коэффицент Стьюдента = 2.8214, соответствует вероятности =

In [10]:
t = stats.t(n-1)
t.cdf(2.8214)

0.9899993808040004

In [11]:
print ('Доверительный интервал равен:', (m - t_int*sigma/np.sqrt(n), m + t_int*sigma/np.sqrt(n)))

Доверительный интервал равен: (194.5265978950862, 202.4734021049138)


Доверительный интервал, построенный с учётом доверительной вероятности 99%, = (194.5, 202.4), включает значение предполагаемого среднего значения а = 200 => утверждение продавца, что средний вес пачки = 200 гр, верно.

### Задача 3
Дана выборка диаметров подшипников из примера 1 с занятия 5:
```
samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ]
```
Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна `0.0625` при уровне значимости `alpha = 0.05`. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: `H = (n - 1) * sample_variance / variance`, где `n` - число элементов в выборке, `sample_variance` - несмещённая оценка дисперсии, `variance` - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

В качестве ответа запишите 1, если нулевая гипотеза верна, и 0 иначе.